src/PostSharp.Engineering.BuildTools/Utilities/ToolInvocationHelper.cs

```diff
@@ -221,6 +228,11 @@ public static class ToolInvocationHelper
                         }
                         else
                         {
+                            if ( FilterProcessOutput( args.Data ) )
+                            {
```

Comment:
No. You are still duplicating code. Move the `if` and the `retry = true` to your local function.